### PR TITLE
feat(input): input groups style changes

### PR DESCRIPTION
**Pull request by bdryanovski**
Small changes on the `.clr-input-group` to solve small alignment issues.

Simple markup to create `clr-input-group` 

```html
<div class="clr-form-control">
    <label for="select-group" class="clr-control-label">Label</label>
    <div class="clr-control-container">
        <div class="clr-input-wrapper">
            <div class="clr-input-group">
                <select>
                    <option>Option A</option>
                    <option>Option B</option>
                    <option>Option C</option>
                </select>
                <input type="text" id="select-group" placeholder="Enter value here" class="clr-input" />
            </div>
        </div>
        <span class="clr-subtext">Helper Text</span>
    </div>
</div>
```
`.clr-input` element have to be inside `<div class="clr-input-group"><!-- Input --></div>`

Horizontal and vertical styled forms & inputs have a little different behavior - keep in mind when checking the examples below.

HorizontalBeforeDemo

Clearly visible that in some cases the bottom border is not going all the way to the end of the grouped container also the warning icons need a little more space.

After removing the `1rem` from the max-width and the padding-right we gonna get this

HorizontalAfter

There is a [StackBlitz Demo](https://stackblitz.com/edit/clarity-input-group-v3?file=src%2Fstyles.css) 

The removing of the padding-right is already done for the datalist component

```css
// src/clr-angular/forms/styles/_datalist.clarity.scss
clr-datalist-container .clr-input-group {
  padding-right: 0;
}
```

Did not find any reason for this
```css
max-width: calc(100% - 1rem);
```
Except that this was the way the grouped element were styled. Prefer to use the push-right, push-bottom style of rearranging elements. 

This affected some of the other input/elements. Like password and datepicker.
HorizontalBasicBefore

After the changes, things look like this.
HorizontalBasicAfter

The changes can also be view inside the dev app. [`npm run start` ](http://localhost:4200/forms/layout-vertical)

##### Testing URLs
http://localhost:4200/datepicker/css-regression
http://localhost:4200/forms/input-group
http://localhost:4200/forms/layout-vertical
http://localhost:4200/forms/layout-horizontal

## TODO

 * [ ] Update documentation with examples and how to implement
 * [ ] Check Gemini tests

## Backporting
If we want to backport it to version 2 we gonna need some additional clases. [Here](https://stackblitz.com/edit/clr-control-group?file=src%2Fstyles.css) I have an example of Version 2 a bit outdated for the moment will try to update it soon. 

Right now the focus is to make it work for Version 3.  

## PR Type

What kind of change does this PR introduce?

* [x] Feature
* [x] Refactoring (no functional changes, no api changes)
* [x] Documentation content changes

Issue Number: #3778

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Other information

Close: #3778

### CODE REVIEW

File: src/clr-angular/forms/styles/_input.clarity.scss [hunk 1]
```diff
@@ -40,6 +40,7 @@
       clr-forms-invalid-color,
       $clr-use-custom-properties
     );
+    // margin-right: 0.3rem;
```
Comment (bdryanovski) [inline src/clr-angular/forms/styles/_input.clarity.scss]: NOTE to self: Remove this comment if not needed

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,5 +1,22 @@
 {
   "files.exclude": {
     "**/*.element.css.ts": true
-  }
+  },
```
Comment (bdryanovski) [inline .vscode/settings.json]: Find a way to keep this changes for myself

File: src/clr-angular/forms/styles/_input-group.clarity.scss [hunk 3]
```diff
@@ -6,9 +6,10 @@
   .clr-input-group {
     @include form-fields-appearance();
     @include input-border-bottom-animation();
-    padding-right: $clr-forms-baseline * 4;
-    max-width: calc(100% - #{$clr_baselineRem_1});
+    padding-right: 0;
```
Comment (bdryanovski) [inline src/clr-angular/forms/styles/_input-group.clarity.scss]: I could remove the padding-right ?!

File: src/clr-angular/forms/styles/_input-group.clarity.scss [hunk 4]
```diff
@@ -6,8 +6,7 @@
   .clr-input-group {
```
Comment (coryrylan) [inline src/clr-angular/forms/styles/_input-group.clarity.scss:6]: Was the PR marked as a breaking change due to the visual changes needed to get the alignments correct or was it a API change needing a new CSS class?

File: src/clr-angular/forms/styles/_containers.clarity.scss [hunk 5]
```diff
@@ -119,7 +119,11 @@
     .clr-subtext {
       display: inline-block;
       margin-top: $clr-forms-baseline;
-      margin-left: $clr-forms-baseline;
+      margin-left: 1rem; //$clr-forms-baseline;
```
Comment (gnomeontherun) [inline src/clr-angular/forms/styles/_containers.clarity.scss:122]: Use variable

File: src/clr-angular/forms/styles/_input-group.clarity.scss (hunk 4 above)
Comment (gnomeontherun) [reply src/clr-angular/forms/styles/_input-group.clarity.scss:6]: I don't see this as a breaking change. The change here is fixing a problem.

**Pull request merged**